builtin/method_str.py

```diff
@@ -477,3 +477,53 @@ def Call(self, rd):
             return value.Str("".join(parts))
 
         raise AssertionError()
+
+
+class Split(vm._Callable):
+
+    def __init__(self):
+        # type: () -> None
+        pass
+
+    def Call(self, rd):
+        # type: (typed_args.Reader) -> value_t
+        """
+        s.split(sep, count=-1)
+
+        Count behaves like in replace() in that:
+        - A negative `count` => ignore
+        - A positive `count` => there will be at most `count` splits
+        - Exception: `count = 0` is an error
```

Comment:
Hm I think zero and negative should be the same case -- ignored, which is equivalent to zero splits



CPython behaves this way too


```
ysh andy@hoover:~/git/oilshell/oil$ = 'foo'.replace('o', 'x', count=1)
(Str)   'fxo'
ysh andy@hoover:~/git/oilshell/oil$ = 'foo'.replace('o', 'x', count=2)
(Str)   'fxx'
ysh andy@hoover:~/git/oilshell/oil$ = 'foo'.replace('o', 'x', count=0)
(Str)   'foo'
ysh andy@hoover:~/git/oilshell/oil$ = 'foo'.replace('o', 'x', count=-1)
(Str)   'fxx'

```

Reply:
Hmm, yeah, I guess we should allow `count=0`. Although, we probably shouldn't "ignore" like a negative count.

For reference, JS and Python do this:

```
>>> "a;b;c".split(";", maxsplit=0)  # Python
['a;b;c']
> "a;b;c".split(';', 0)  // JS
[]
```

JS doesn't include the "remainder" chunk, so it sort of agrees with Python:

```
> "a,b,c".split(',', 1)
[ 'a' ]
> "a,b,c".split(',', 2)
[ 'a', 'b' ]
```

But we follow Python's style of including a "remainder" elsewhere, so I'll do that for `count=0` too:

```
ysh$ = 'a;b;c'.split(';', count=0)
(List)   ["a;b;c"]
```